utils/src/main/kotlin/LicenseFilenamePatterns.kt

```diff
@@ -44,4 +46,65 @@ object LicenseFilenamePatterns {
     val ROOT_LICENSE_FILENAMES = listOf(
         "readme*"
     ).generateCapitalizationVariants()
+
+    /**
+     * Return glob patterns which match all files which may contain license information residing recursively within the
+     * given absolute [directory] or in any of its ancestor directories.
+     */
+    fun getLicenseFileGlobsForDirectory(directory: String): List<String> =
+        globFileInDirectoryOrAncestors(directory, LICENSE_FILENAMES + ROOT_LICENSE_FILENAMES)
+
+    /**
+     * Return recursively all ancestor directories of the given absolute [directory].
+     */
+    private fun getAllAncestorDirectories(directory: String): List<String> {
+        val result = mutableListOf<String>()
+
+        var ancestorDir = File(directory).parentFile
+        while (ancestorDir != null) {
+            result.add(ancestorDir.invariantSeparatorsPath)
+            ancestorDir = ancestorDir.parentFile
+        }
+
+        return result
+    }
+
+    /**
+     * Return a glob pattern which matches files in [directory], in any ancestor directory of [directory] and
+     * optionally (if [matchSubDirs] is true) recursively in any sub-directory of directory, if the filename
+     * matches the [filenamePattern].
+     */
+    internal fun globFileInDirectory(
```

Comment:
The function name is unclear to me, what about `getFileGlobForDirectory`?

Reply:
Done.